src/session/content/chat_action_bar.rs

```diff
@@ -287,13 +291,218 @@ impl ChatActionBar {
         self.save_message_as_draft();
 
         let imp = self.imp();
+        let mut bindings = imp.bindings.borrow_mut();
+        while let Some(binding) = bindings.pop() {
+            binding.unwatch();
+        }
 
         if let Some(ref chat) = chat {
             self.load_draft_message(chat.draft_message());
 
             imp.chat_action_in_cooldown.set(false);
+
+            let chat_expression = gtk::ConstantExpression::new(chat).upcast();
+            let permissions_expression = gtk::PropertyExpression::new(
+                Chat::static_type(),
+                Some(chat_expression.clone()),
+                "permissions",
+            )
+            .upcast();
+
+            // Handle whether or not message bar should be shown
```

Comment:
I wasn't sure if I should make it one huge expression, so I made four of them for all chat types

Reply:
I guess it's "fine", in the sense that gtk bindings are really complicated to use and become really big too easily. Anyway, I think that some stuff could still be simplified, I'll make a proper review soon.